### PR TITLE
British Library Universal Viewer changes

### PR DESCRIPTION
**Pull request by edsilv**
These are a few changes I needed to make for the British Library Universal Viewer project.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ build/
 sftp-config.json
 coverage/
 temp/
+.idea
```
Comment (iangilman) [inline .gitignore]: Do we need this?

File: src/viewer.js [hunk 2]
```diff
@@ -835,6 +835,13 @@ $.extend( $.Viewer.prototype, $.EventSource.prototype, $.ControlDock.prototype,
         return this;
     },
 
+    hideControls: function() {
+        beginControlsAutoHide(this);
+    },
+
+    showControls: function() {
+        abortControlsAutoHide(this);
+    },
```
Comment (iangilman) [inline src/viewer.js]: Why not use Viewer.setControlsEnabled()?

File: src/navigator.js [hunk 3]
```diff
@@ -282,7 +282,7 @@ $.extend( $.Navigator.prototype, $.EventSource.prototype, $.Viewer.prototype, /*
             bottomright;
 
         viewerSize = $.getElementSize( this.viewer.element );
-        if ( this._resizeWithViewer && !viewerSize.equals( this.oldViewerSize ) ) {
+        if ( this._resizeWithViewer && viewerSize.x && viewerSize.y && !viewerSize.equals( this.oldViewerSize ) ) {
```
Comment (iangilman) [inline src/navigator.js]: Good catch!

**Comment by iangilman** [comment]
Thank you...just a couple questions.

File: .gitignore (hunk 1 above)
Comment (edsilv) [inline .gitignore]: Probably not, just useful for WebStorm users working on a fork of OSD.

File: src/viewer.js (hunk 2 above)
Comment (edsilv) [inline src/viewer.js]: Good point, removed.

File: .gitignore (hunk 1 above)
Comment (iangilman) [inline .gitignore]: Fair enough; we can keep it.

**Comment by iangilman** [comment]
Thank you!

**Pull request merged**